frontend/src/scenes/persons/PersonHeader.tsx

```diff
@@ -4,18 +4,24 @@ import './PersonHeader.scss'
 import { Link } from 'lib/components/Link'
 import { urls } from 'scenes/urls'
 import { ProfilePicture } from 'lib/components/ProfilePicture'
+import { useValues } from 'kea'
+import { teamLogic } from 'scenes/teamLogic'
 
 export interface PersonHeaderProps {
     person?: Partial<Pick<PersonType, 'properties' | 'distinct_ids'>> | null
     withIcon?: boolean
     noLink?: boolean
 }
 
-export const asDisplay = (person: Partial<PersonType> | PersonActorType | null | undefined): string => {
+export const asDisplay = (
+    person: Partial<PersonType> | PersonActorType | null | undefined,
+    displayNameProperties: string[] = ['email', 'name', 'username']
+): string => {
```

Comment:
At first I thought it might be a big simplification for all downstream code to just pass `team` to this function: `asDisplay(person, team)`. It'll make code simpler, and those two objects are the ones that are linked in the end. It'll also be easier to make the name depend on other team properties later on... if needed.

However, the `team` does not change at all during the lifetime of the browser session. It's a design and safety feature against accidental data leakage (E.g. a rouge property picker or a random insight `cache`).

That means, with 99.999% certainty, whenever `asDisplay` is called, there will also be a mounted `teamLogic` around, and it will not change.

Knowing this, you can use `findMounted()` to find it, and fall back to the defaults:

```ts
const team = teamLogic.findMounded()?.values?.currentTeam
const personDisplayNameProperties =  team?.person_display_name_properties ?? ['email', 'name', 'username']
```

directly in the body of `asDisplay`.

Reply:
Basically, yeah.

You could also just write `teamLogic.values.currentTeam`, but that'll just throw if the logic isn't mounted. It'll be most of the time, but let's be sure. You might not care about it in e.g. tests.

Whether a logic is mounted or not is controlled by React: whether we're rendering a component that requests a logic or not. `App` always mounts `teamLogic`: https://github.com/PostHog/posthog/blob/f0861deb24d499838d2cee04d2f62b4dd0ab92c1/frontend/src/scenes/App.tsx#L63-L67

The downside is that if ever `currentTeam` changes, the value won't autoupdate, like it would with the custom hook. But it won't change :).